onnxruntime/core/providers/nnapi/nnapi_builtin/builders/impl/resize_op_builder.cc

```diff
@@ -81,8 +81,11 @@ Status ResizeOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const
   const auto& output = node_unit.Outputs()[0].node_arg.Name();
 
   auto input = inputs[0].node_arg.Name();
-  bool use_nchw = model_builder.UseNCHW();
-  ORT_RETURN_IF_ERROR(IsOpInRequiredLayout(use_nchw, node_unit));
+
+  const auto& output_shape = shaper[output];
+  const auto& input_shape = shaper[input];
+
+  const bool use_nchw = output_shape[3] != input_shape[3];  // not Channel last
```

Comment:
nit: would be clearer to call this 'input_is_nchw' as the UseNCHW is an explicit option for the EP that has a slightly different meaning.

it would also be more consistent with the checks in ResizeOpBuilder::IsOpSupportedImpl to do the test on dim 1 here.

Reply:
Good suggestion. Done